### DISCUSSION TITLE
Avoid resend sms at page reload

### DISCUSSION THREAD

**Discussion by evrenbayrak** [dasniko/keycloak-2fa-sms-authenticator#35] [Q&A]
How to avoid resend sms while user manually reload web page? Every reload triggers authenticate method and i think there must be some state check for avoid this.

**Comment by dasniko** [comment]
Sure.
This repo is not a ready-built-and-run-out-of-the-box solution.
It's an example of how one can implement such a solution.
I have an advanced version of this extension running successfully at various customers.

**Reply by evrenbayrak** [reply]
Thanks for the reply, 
What about moving generate otp code and send sms operation to the 'action' function? This might be confusing but it will work depending on the form parameter.
Do you think it's the right approach?
